js/widgets.js

```diff
@@ -1,27 +1,97 @@
+        return arrays.map((array) => array[i]);
+    });
+}
+function buildIntegrationFragment(reference) {
+    const [kind, ...ids] = reference.split('-');
+    const id = ids.join('-');
+    const fragment = document.createElement('div');
+    const help = document.createElement('p');
+    help.innerHTML = `<a href="#">Copy-paste</a> that code into your
+    own HTML and/or <a href="${baseURL}/faq/developer/">read the docs</a>.`;
+    fragment.appendChild(help);
+    const textarea = document.createElement('textarea');
+    textarea.style.width = '150%';
+    textarea.innerHTML = `<div data-udata-${kind}-id="${id}"></div>
+<script src="${scriptURI}" id="udata" async defer></script>`;
+    fragment.appendChild(textarea);
+    return fragment;
+}
+function easeCopyPasting(content) {
+    const textarea = content.lastChild;
+    textarea.focus();
+    textarea.select();
+    const help = content.firstChild;
+    help.firstChild.addEventListener('click', (e) => {
+        e.preventDefault();
+        textarea.select();
+        document.execCommand('copy');
+    });
+}
+function handleIntegration(event) {
+    event.preventDefault();
+    const element = event.target;
+    const aside = element.parentNode.parentNode;
+    const content = aside.previousElementSibling;
+    const article = aside.parentNode;
+    const fragment = buildIntegrationFragment(article.id);
+    content.innerHTML = fragment.innerHTML;
+    easeCopyPasting(content);
+}
+
 // Warning: using `Array.from` adds 700 lines once converted through Babel.
-[].slice.call(document.querySelectorAll('[data-udata-dataset-id]')).forEach((el) => {
-    const datasetId = el.getAttribute('data-udata-dataset-id');
-    const oEmbedObjectURL = window.encodeURIComponent(
-        `${window.udataURL}/api/1/datasets/${datasetId}/`);
-    const url = `${window.udataURL}/api/1/oembed/?url=${oEmbedObjectURL}`;
+const territories = [].slice.call(document.querySelectorAll('[data-udata-territory-id]'));
+const datasets = [].slice.call(document.querySelectorAll('[data-udata-dataset-id]'));
+// Extract the base URL from the URL to that script.
+const parser = document.createElement('a');
+const scriptURI = document.querySelector('script#udata').src
+parser.href = scriptURI;
+const baseURL = `${parser.protocol}//${parser.host}`;
+// Keep the chunk > 9 otherwise territories pages will issue more than one query.
+territories.concat(datasets).chunk(9).forEach((elements) => {
+    const references = elements.map((el) => {
+        if (el.hasAttribute('data-udata-territory-id')) {
+            const territoryId = el.getAttribute('data-udata-territory-id');
+            return `territory-${territoryId}`;
+        } else if (el.hasAttribute('data-udata-dataset-id')) {
+            const datasetId = el.getAttribute('data-udata-dataset-id');
+            return `dataset-${datasetId}`;
+        }
+    });
+    const url = `${baseURL}/api/1/oembeds/?references=${references}`;
     fetch(url)
```

Comment:
We might need, compatibility is too short: https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility